src/main.c

```diff
@@ -2666,14 +2735,22 @@ static int on_config_listen_element(h2o_configurator_command_t *cmd, h2o_configu
             return -1;
         if (rcvbuf_node != NULL && h2o_configurator_scanf(cmd, *rcvbuf_node, "%u", &stream_rcvbuf) != 0)
             return -1;
+        if (reconnect_interval_node != NULL && h2o_configurator_scanf(cmd, *reconnect_interval_node, "%lu", &reconnect_interval) != 0)
+            return -1;
+        if (connections_per_thread_node != NULL && h2o_configurator_scanf(cmd, *connections_per_thread_node, "%lu", &connections_per_thread) != 0)
+            return -1;
```

Comment:
Here also? Though I think we'd be fine using `int` instead (see max-connections).

Reply:
Addressed in https://github.com/h2o/h2o/pull/3405/commits/a512ac174240ded47c5ed5546882a017ef22649e. 

>  I think we'd be fine using int instead (see max-connections).

This is something I've always wondered about. I understand that 64bit doesn't need to be specified, but why use int instead of unsigned? Are negative values expected?

Reply:
I think it comes from the fact that `sync_fetch_and_add` is a pointer and a value of same type, and that we want to pass `-1` as the second argument (see how `num_connections(-1)` is used).

If we change `num_connections` to an unsigned type, we would have to worry how `-1` is promoted to an unsigned type (and that expansion does not happen after the sign conversion). That's a needless complexity IMO.